lambda-deployment-orca/src/main/java/com/amazon/aws/spinnaker/plugin/lambda/verify/LambdaCacheRefreshTask.java

```diff
@@ -44,33 +50,108 @@
+                            case 200:
+                                return true;
+                            // Async processing... meaning we have to poll to see when the refresh finished....
+                            case 202:
+                                onDemandRefreshResponse = objectMapper.readValue(respString, OnDemandResponse.class);
+                                if (StringUtils.isEmpty(onDemandRefreshResponse.getCachedIdentifiersByType().get("onDemand").isEmpty())) {
+                                    throw new NotFoundException("Force cache refresh did not return the id of the cache to run.  We failed or cache refresh isn't working as expected");
+                                }
+                                String id = onDemandRefreshResponse.getCachedIdentifiersByType().get("onDemand").get(0);
+                                waitForCacheToComplete(id, startTime, props.getCloudDriverBaseUrl(), 30);
+                                return true;
+                            default:
+                                logger.warn("Failed to generate a force cache refresh with response code of " + response.code() + "... retrying.");
+                                throw new RuntimeException("Failed to process cache request due to an invalid response code... retrying...");
+                        }
+                    } catch (IOException e) {
+                        throw new RuntimeException("Error communicating with clouddriver", e);
+                    }
+                }, tries, Duration.ofSeconds(15), false
+        );
+
+    }
+
+    /*
+        IF no data, trigger a force refresh
+        if processedCount > 0 && processedTime > stageStart
+            SUCCESS
+        ELSE
+            TRIGGER A REFRESH
+     */
+    private void waitForCacheToComplete(String id, long taskStartTime, String cloudDriverUrl, int retries) {
+
+        new RetrySupport().retry(
+                () -> {
+                    try {
+                        String response = utils.getFromCloudDriver(cloudDriverUrl + "/cache/aws/function?id=" + id);
+                        Collection<Map<String, Object>> onDemands = objectMapper.readValue(response, Collection.class);
+                        for (Map<String, Object> results : onDemands) {
+                            if ((int) results.getOrDefault("processedCount", 0) > 0 && (long) results.getOrDefault("cacheTime", 0) > taskStartTime) {
+                                logger.info("Caching should be completed for " + id);
+                                return true;
+                            } else {
+                                throw new RuntimeException("No cache data has been processed... retrying...");
+                            }
+                        }
+                        logger.warn("No on demand cache refresh found for  " + id);
+                        throw new RuntimeException("No on demand cache refresh found for " + id);
```

Comment:
same as above

Reply:
Same as above - how the Retry system works if you want to trigger a retry.